lib/Parser/Parse.cpp

```diff
@@ -9063,6 +9063,16 @@ ParseNodePtr Parser::ParseExpr(int oplMin,
             // ArrowFunction/AsyncArrowFunction is part of AssignmentExpression, which should terminate the expression unless followed by a comma
             if (m_token.tk != tkComma)
             {
+                if (!(m_token.tk == tkRCurly ||
+                    m_token.tk == tkRParen ||
+                    m_token.tk == tkRBrack ||
+                    m_token.tk == tkSColon ||
+                    m_token.tk == tkLimKwd ||
+                    m_token.tk == tkEOF ||
+                    m_pscan->FHadNewLine()))
```

Comment:
This seems kind of fragile - I was hoping we had something a bit more general to detect the valid terminators, but I don't think we do. This is ok for now.